docs/guides/advanced/fee_model.md

```diff
@@ -12,23 +12,134 @@ The gas fee covers the following expenses:
 - Sending 'bytecode' to L1 (if not already there) - typically a one-time cost when deploying a new contract
 - Closing the batch and handling proofs - This aspect also relies on L1 costs (since proof publication must be covered).
 
+## Price configuration
+
+We have two pricing models:
+
+- the 'L1Pegged' - until protocol version 19
+- the 'PubdataIndependent' - from protocol version 20 (release 1.4.1)
+
+### L1 pegged ('old' fee model)
+
+Under this fee model, operator was providing `FeeParamsV1`, which contained:
+
+- l1_gas_price
+- minimal_l2_gas_price
+
+then, the system was computing `L1PeggedBatchFeeModelInput`, that contained
+
+- l1_gas_price
+- 'fair' l2 gas price - which in 99% of cases was equal to minimal_gas_price, and was greater from it only if l1 gas
+  price was huge, to guarantee that we can publish enough data in each transaction (see table below for details).
+
+Many other values, were 'hardcoded' within the system (for example how to compute the pubdata price based on l1 gas
+price, how much committing the proof to L1 costs etc).
+
+### PubdataIndependent ('new' fee model)
+
+This method is called `PubdataIndependent` and the change was done to allow more flexibility in pubdata costs (for
+example if pubdata is published to another Data Availability layer, or it not published at all - in case of validium).
+
+In this model, there are 8 config options, let's walk through them:
+
+`FeeParamsV2` contains 2 dynamic prices:
+
+- `l1_gas_price` - which is used to compute the cost of submitting the proofs on L1
+- `l1_pubdata_price` - which is the cost of submitting a single byte of pubdata
+
+And config options (`FeeModelConfigV2`) contain:
+
+- `minimal_l2_gas_price` - same as above, this should cover the $ costs of running the system
+
+2 fields around the maximum capacity of the batch:
+
+- `max_gas_per_batch` - maximum amount of gas in a batch - if you set this too high, you might exceed the circuits
+  limit.
+- `max_pubdata_per_batch` - the maximum amount of pubdata that we can put in a single batch (usually it is around 128kb
```

Comment:
nit: I'd specify that this limitation is due to space available for pubdata publishing on Ethereum